### ISSUE TITLE
There was a problem executing the code. The files myTileDataset.py and myPromptTileDataset.py could not be found.

### ISSUE THREAD

**Issue by Snaillms** [7LFB/QAP#2]
Dear author,

I recently read your paper and was very interested in your field. I want to reproduce the code, but I encountered some difficulties. I did not find the following 4 files. Can you provide these codes? I will be grateful.

  - myTileDataset.py
  - myPromptTileDataset.py
  - mySlideDataset.py（ This can find the class with this name）
  - myPromptSlide.py（This file doesn't seem to be used.）

Thanks！

**Comment by 7LFB** [comment]
Thank you for your interest in our work. We apologize for any confusion in the readme file. In the 'myDatasets' folder, there are two custom dataset loaders: myPromptAugTileDataset.py for tile processing and myPromptSlideDataset.py for WSI processing. That's all the dataloaders for training and testing.  If you have any questions, feel free to contact me.

**Comment by Snaillms** [comment]
OK, thank you very much. You only have model_v36.py in the models folder, but I can't find other versions of model, and in your testSlide.py file, there is an error like "Cannot resolve import "models.model_v31"". I want to know whether I should create the missing py file, and what is the difference between model_v36 and other versions, so that I can restore the missing py file. Of course, if you can provide your file, I will be grateful.
![image](https://github.com/user-attachments/assets/7e5c2d23-bd8e-47b9-9e31-c71787cf2c1f)

**Comment by 7LFB** [comment]
model_v36 serves as our default version and is the one used in our published paper, while the other versions are outdated.

**Comment by Snaillms** [comment]
Thank you! I wish you good luck in your research.

**Comment by Snaillms** [comment]
Dear author, I would like to know if the suffix of this file (vit_256_small_dino.pth.txt) is available? The content of the file is empty.
I encountered a problem when loading the model file. The information I searched online said: Make sure the file you load is indeed a PyTorch model file (usually a .pth or .pt file), so I would like to know if the suffix of your pre-trained file is correct?

Secondly, what version of pytorch did you use in this project?
I hope to get your reply again, thank you very much! !
![image](https://github.com/user-attachments/assets/db92df98-040b-4ec0-8efa-db23df1efde0)

**Comment by Snaillms** [comment]
Dear author, I would like to know if the suffix of this file (vit_256_small_dino.pth.txt) is available? The content of the file is empty.
I encountered a problem when loading the model file. The information I searched online said: Make sure the file you load is indeed a PyTorch model file (usually a .pth or .pt file), so I would like to know if the suffix of your pre-trained file is correct?

Secondly, what version of pytorch did you use in this project?
I hope to get your reply again, thank you very much! !

![1733835352354](https://github.com/user-attachments/assets/f010d5b5-b10b-4138-8d5e-034cda33ce7c)

**Comment by 7LFB** [comment]
Yes, you are right. You can download the pretrained checkpoints here: https://github.com/mahmoodlab/HIPT/tree/master/HIPT_4K/Checkpoints
Pytorch version: 2.4

**Comment by Snaillms** [comment]
Dear author, I keep encountering library function version compatibility issues when running the project. Can you provide a script (install.sh) or requirements.txt to install the dependent library?

**Comment by 7LFB** [comment]
I have create a requirements.txt file. Feel free to contact me if there are any problem. Thanks.

**Comment by Snaillms** [comment]
I am sorry to bother you again. I keep getting errors when running the code. The prompt is that the DGL and torch versions do not match. I would like to ask which version of DGL are you using? I am very sorry.

**Comment by 7LFB** [comment]
Sorry for that. I installed DGL==1.1.2. It seems that I did not use the lib
in this code. Could you tell me more details about it?

Snaillms ***@***.***> 于2024年12月12日周四 21:58写道：

**Comment by Snaillms** [comment]
Sorry, I originally ran the code after installing other dependencies, but it prompted that there was no DGL dependency. Then I found that it was not available to install this library directly using the pip command. Later, I found that I needed to go to the DGL official website to select the corresponding configuration to download. Finally, the download was successful and it can be used. I think this problem should not appear again for the time being. Thank you!

**Comment by Snaillms** [comment]
Dear author,
1. In your experiment, you used the dataset Liver-NAS, which comes from the paper "Screening for compensated advanced
chronic liver disease using refined baveno vi elastography cutoffs in asian patients with nonalcoholic fatty liver disease.". I browsed this paper and did not find the direct dataset. My understanding is that this dataset is one of the final results of this article?
2. Can you provide this dataset? Do you have no right to make this dataset public? Do I need to find the author of the paper to obtain this dataset myself?
3. Before conducting the training experiment, I must find this dataset for data processing. Is this idea correct?
This experiment is very important to me. I look forward to your reply!

**Comment by 7LFB** [comment]
We have conducted three tasks with two datasets. For NAFLD-Anomaly, you can
download here: https://osf.io/gqutd/
<https://github.com/7LFB/QAP/issues/url>
For Liver-NAS, it's a private dataset which is not publicly available yet.

Snaillms ***@***.***> 于2024年12月13日周五 20:50写道：

**Comment by Snaillms** [comment]
OK, Thank you!
I wish you good luck in your research！

**Comment by Snaillms** [comment]
Dear author:
I would like to know at which steps (k function, image segmentation and other preprocessing) the three folders nuclei-kfunc, white_segment, and nuclei_segment_watershed are generated? Is there a py file for processing this part?

Are the folders LiverBiopsyPatches40X224, LiverBiopsyPatches40X224Prior, and LiverBiopsyExcels that appear in testSlide.py the folders generated after preprocessing?

Looking forward to your reply again！

**Comment by 7LFB** [comment]
The process is like this:
- Step-1:
     Extract nuclei and white region segmentation
- Step-2:
Extract properties or attributes (e.g., k-function, histogram) from the segmentation.

Then we can use these attributes to build visual prompts.
Step-1&2 can be achieved in online manner by setting
- prior_nuclei=1 (default=0) # time consuming
- prior_white=1 (default=1) # not time consuming
- generate_props=1 (default=1) # not time consuming

Otherwise, we need to process it in advance and save the results to the corresponding folder, and set the path to 
- nuclei_folder = 'path/to/your/nuclei segmentation'
- white_folder='path/to/your/white segmentation'

As for the folder, 
LiverBiopsyPatches40X224 # path to your image dataset
LiverBiopsyPatches40X224Prior # path to your prior segmentation results if it is processed offline
LiverBiopsyExcels # path to your data list

Thanks.

**Comment by Snaillms** [comment]
Thank you! I will try it later.
